cuda/preconditioner/jacobi_advanced_apply_kernels.cu

```diff
@@ -7,7 +7,7 @@
 #include "core/matrix/dense_kernels.hpp"
 #include "core/preconditioner/jacobi_kernels.hpp"
 #include "core/synthesizer/implementation_selection.hpp"
-#include "cuda/preconditioner/jacobi_common.hpp"
+#include "preconditioner/jacobi_common.hpp"
```

Comment:
our header search path are all from the top level

Reply:
They don't need to be - the jacobi_common headers differ slighly between CUDA and HIP, and are the only generated headers in the backend, so I believe we can treat them differently from the other headers. There might be a more stable solution, but I didn't attempt to find it for now. Alternatively we could attempt to do weird preprocessor things like `#include #GKO_DEVICE_NAMESPACE "preconditioner/jacobi_common.hpp"`, but I am worried about how portable that would be

Reply:
I'll revert the Jacobi-related changes for now, it's probably better to do this together with the other Jacobi kernels anyways.

Reply:
Found a better solution - we can mirror the directory structure inside the `cuda` and `hip` build directory